integration-tests/src/tests/standard_cases/mod.rs

```diff
@@ -1348,15 +1362,78 @@ pub fn test_contract_write_key_value_cost(node: impl Node) {
         assert_eq!(node_touches, results[i]);
     }
 }
+
+fn make_write_key_value_receipts(node: &impl Node) -> Vec<Receipt> {
+    (0..3)
+        .map(|i| {
+            let receipt_enum = ReceiptEnum::Action(ActionReceipt {
+                signer_id: alice_account(),
+                signer_public_key: node.signer().as_ref().public_key(),
+                gas_price: 0,
+                output_data_receivers: vec![],
+                input_data_ids: vec![],
+                actions: vec![FunctionCallAction {
+                    method_name: "write_key_value".to_string(),
+                    args: arr_u64_to_u8(&[i, 10u64 + i]),
+                    gas: 10u64.pow(14),
+                    deposit: 0,
+                }
+                .into()],
+            });
+            Receipt {
+                predecessor_id: alice_account(),
+                receiver_id: bob_account(),
+                receipt_id: hash(&receipt_enum.try_to_vec().unwrap()),
+                receipt: receipt_enum,
+            }
+        })
+        .collect()
+}
+
+/// Check correctness of touching trie node cost for receipts with enabled chunk nodes cache.
+/// We run the same set of receipts 2 times and compare resulting costs. Each receipt writes some key-value pair to the
+/// contract storage.
+/// 1st run establishes the trie structure. For our needs, the structure is:
+///
+///                                                    --> (Leaf) -> (Value 1)
+/// (Extension) -> (Branch) -> (Extension) -> (Branch) |-> (Leaf) -> (Value 2)
+///                                                    --> (Leaf) -> (Value 3)
+///
+/// 2nd run should count 6 nodes for the first receipt and 2 nodes for next two receipts, because for them first 4 nodes
+/// were already put into the chunk cache.
+pub fn test_chunk_nodes_cache_across_receipts(node: impl Node, runtime_config: RuntimeConfig) {
+    let node_user = node.user();
+    let mut node_touches: Vec<u64> = vec![];
+    #[cfg(feature = "protocol_feature_chunk_nodes_cache")]
+    let results: Vec<u64> = vec![6, 2, 2];
+    #[cfg(not(feature = "protocol_feature_chunk_nodes_cache"))]
+    let results: Vec<u64> = vec![6, 6, 6];
+    for i in 0..2 {
+        let receipts = make_write_key_value_receipts(&node);
+        let receipt_hashes: Vec<CryptoHash> =
+            receipts.iter().map(|receipt| receipt.receipt_id.clone()).collect();
+
+        node_user.add_receipts(receipts).unwrap();
+
+        if i == 1 {
+            node_touches = receipt_hashes
+                .iter()
+                .map(|receipt_hash| {
+                    let result = node_user.get_transaction_result(receipt_hash);
+                    let touching_trie_node_cost = get_touching_trie_node_cost(&result.metadata);
+                    touching_trie_node_cost
+                        / runtime_config.wasm_config.ext_costs.touching_trie_node
+                })
+                .collect();
+        }
```

Comment:
as we are 'disabling' the cache directly after each runner call:

 do we have a test that checks that the cost of accessing the nodes 'outside' of the runtime is still counted?

Reply:
Hm, could you clarify what 'outside' means?
We don't charge these costs outside of `Runtime` or `NightshadeRuntime` (and even `action_function_call` scope)